sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/ProfileContainerData.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.resources.Resource;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A ProfileContainer represents a single profile. It wraps pprof profile with OpenTelemetry
+ * specific metadata.
+ *
+ * @see "profiles.proto::ProfileContainer"
+ */
+@Immutable
+public interface ProfileContainerData {
+
+  /** Returns the resource of this profile. */
+  Resource getResource();
+
+  /** Returns the instrumentation scope that generated this profile. */
+  InstrumentationScopeInfo getInstrumentationScopeInfo();
+
+  /**
+   * Returns a globally unique identifier for a profile. The ID is a 16-byte array. An ID with all
+   * zeroes is considered invalid. This field is required.
+   */
+  @SuppressWarnings("mutable")
+  byte[] getProfileId();
```

Comment:
maybe this should be a String, like we use for traceids and spanids.

Reply:
what is responsible for generating this id?

Reply:
Trying to get to the bottom of this one. Not entirely sure the way spanid/traceid is handled is such a good example, I'm finding it difficult to follow...
SpanContext offers both `String getTraceId()` (32 char hex) and `byte[] getTraceIdBytes()` (16 bytes, via OtelEncodingUtils.bytesFromBase16)  The traces wire spec says the Span.trace_id proto field is 16 bytes, so what the marshallers should be using is getTraceIdBytes. But what they actually use is getTraceId. Which would still be ok, if they then converted with OtelEncodingUtils. But they appear not to, instead packing a 32 byte array with the string utf8? I think the usability issue is that in general a String is expected to be utf8 encoded on the wire, but trace/span are (or should be) hex converted first, effectively overloading the semantics of 'String' in the interface definitions.
Anyhow, for profiles I'm happy  to go with adding a String form getter alongside the byte[] one. As to which getter the marshaller should use, I think byte[] and leave the conversion to the class that's implementing the interface method, but having the marshaller call OtelEncodingUtils on the String would be equally valid, just duplicating functionality the impl class would need to have anyhow.

Reply:
Treating this the same as spanid/traceid/profileid in making the String a hex representation of the arbitrary bytes, rather than UTF8 representation.

Reply:
> The traces wire spec says the Span.trace_id proto field is 16 bytes, so what the marshallers should be using is getTraceIdBytes. But what they actually use is getTraceId. Which would still be ok, if they then converted with OtelEncodingUtils.

This is what's happening: https://github.com/open-telemetry/opentelemetry-java/blob/2e3e04c7e882722c52e1c56e4e8604e78594d87d/exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/ProtoSerializer.java#L38-L39